### ISSUE TITLE
The second call causes double-overlayed Video Streams on the person's device that first joined the room

### ISSUE THREAD

**Issue by AgentKnopf** [twilio/video-quickstart-android#49]
Summary: When attempting to make a second call to a different room than before the user who first connects to the room has both streams rendered in his main window: local and remote

### How to reproduce

- This test uses two rooms: zebra and dolphin (you can name them whatever you want)
- Authenticate on both devices
- Have both users join the room "zebra"
- Once the second person joins both will receive each other's video feed
- Disconnect the call on one device
- The call is automatically disconnected on the other device as well (both disconnect from the room)
- Now one device connects to the room "dolphin"
- The second device joins that room as well
- The call is established and both can see each others video streams
- **Problem**: The first person that joined the room, receives two overlaying streams in his main window: His local stream and the remote stream. In my case the double-overlayed stream occurs on the device that first connected to the room

### Video recording of how it looks

https://youtu.be/AbdT6snUUds

### Devices used for the test:
- Google Pixel with Android 7.1.1
- Asus Nexus Tablet with Android 6.1
- Sony Xperia Z5 Compact (in combination with Google Pixel) with Android 6.x

Let me know if you need more information. I managed to reproduce this issue each time, following above steps.

**Comment by aaalaniz** [comment]
Is this issue reproducible in the vanilla quick start application or in your own application? I will try to reproduce locally. I recommend making sure that you remove the primary renderer from your local video track. That would explain the double rendering.

**Comment by AgentKnopf** [comment]
@aaalaniz I tried it with the same version that I posted in the Sony Xperia S bug report - it didn't have any modification, except how the token is obtained, so I'd think you should be able to reproduce it as well. The video unfortunately has been removed by youtube, but I had a comment on there from someone else who had the same issue on other devices (very likely has nothing to do with what device you run it on - as you indicated in your comment). If you can't reproduce it, I'll give it another go with Twilio's vanilla version.

**Comment by aaalaniz** [comment]
@AgentKnopf I have reproduced locally. Will have a fix soon. Thanks for the very details steps.

**Comment by aaalaniz** [comment]
@AgentKnopf please try out #50 and see if this issue is resolved. I will merge once validated. Thanks!

**Comment by AgentKnopf** [comment]
@aaalaniz will check it out once back home - in about 12 hours - thanks a ton for the quick fix :) !

**Comment by Schwonsen** [comment]
@aaalaniz after your fix the bug is still there but now this double rendering issue already starts on the first entering of a room + the primaryVideoView of the second device which enters a room always shows his localVideoTrack.

**Comment by AgentKnopf** [comment]
@aaalaniz I integrated your changes into my project and it works correctly now, I can no longer reproduce the issue. I connected to various room joining combinations in one session such as:

- Dolphin
- Zebra
- Dolphin
- Dolphin

In no combination the originally reported error occurred. To be fair though: I merged your Pull Request changes into my own repository, which however is identical to the original, except for the way the token is obtained. So I am quite sure your fix works.

@Schwonsen If you are still experiencing problems maybe check if you merged the changes correctly and if that is the case provide some step-by-step guide to reproduce the issue you're experiencing.
